src/shim/kernel.rs

```diff
@@ -17,7 +17,7 @@ macro_rules! error_number {
         #[derive(Debug, Clone)]
         #[non_exhaustive]
         pub enum ErrorNumber {
```

Comment:
I don't think it should be - enum layout is only useful for:
- casting to primitives. This can never be cast since it's got a non-primitive variant, the unknown one
- FFI layout

It would also be incorrect - the fvm `#[repr]`s correspond to the actual error code, but if anyone read the discriminant, it would be incorrect in the unknown case.